.github/workflows/ci-workflow.yml

```diff
@@ -10,31 +10,19 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-
-      - name: Install pnpm
-        uses: pnpm/action-setup@v2.2.4
+      - uses: pnpm/action-setup@v2
         with:
           version: 8
-
-      - name: Read .nvmrc
-        run: echo "##[set-output name=NVMRC;]$(cat .nvmrc)"
-        id: nvm
-
-      - name: Setup Node
-        uses: actions/setup-node@v3
+      - uses: actions/setup-node@v3
         with:
-          node-version: ${{ steps.nvm.outputs.NVMRC }}
-          check-latest: true
-
-      - name: Install dependencies
-        run: pnpm install
+          cache: 'pnpm'
+          cache-dependency-path: 'pnpm-lock.yaml'
```

Comment:
I used cache, cache-dependency-path to cache 
installed dependencies

https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

This action follows [actions/cache](https://github.com/actions/cache/blob/main/examples.md#node---npm) guidelines,